lib/vehicle_factory.rb

```diff
@@ -0,0 +1,28 @@
+class VehicleFactory
+  attr_reader :created_vehicles 
+  
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles(vehicle_details)
```

Comment:
This method should be responsible for creating new `Vehicle` objects. I am assuming that `vehicle_details` is an array of data. So you can just iterate over this and instantiate a new `Vehicle` and pass it an argument that is a hash with the attributes that you for each vehicle object. 
```ruby
def create_vehicles(vehicle_details)
   vehicle_details.map do |vehicle|
     Vehicle.new({
       engine: :ev,
       make: vehicle[:make],
       model: vehicle[:model],
       vin: vehicle[:vin_1_10],
       year: vehicle[:model_year]
     })
end
```
This returns a new array.